src/Mod/PartDesign/App/FeatureExtrude.cpp

```diff
@@ -480,18 +480,36 @@ App::DocumentObjectExecReturn* FeatureExtrude::buildExtrusion(ExtrudeOptions opt
 
     // Validate parameters
     double L = Length.getValue();
-    if ((method == "Length") && (L < Precision::Confusion())) {
-        return new App::DocumentObjectExecReturn(
-            QT_TRANSLATE_NOOP("Exception", "Length too small"));
+    if (method == "Length") {
+        if (std::abs(L) < Precision::Confusion()) {
```

Comment:
This case should also be included in `if (std::abs(L + L2) < Precision::Confusion())`